internal/datacoord/snapshot.go

```diff
@@ -0,0 +1,1471 @@
+	if stats == nil {
+		return nil
+	}
+	return &AvroJsonKeyStats{
+		FieldID:                stats.GetFieldID(),
+		Version:                stats.GetVersion(),
+		Files:                  stats.GetFiles(),
+		LogSize:                stats.GetLogSize(),
+		MemorySize:             stats.GetMemorySize(),
+		BuildID:                stats.GetBuildID(),
+		JsonKeyStatsDataFormat: stats.GetJsonKeyStatsDataFormat(),
+	}
+}
+
+// convertAvroToJsonKeyStats converts Avro JsonKeyStats back to protobuf format.
+func convertAvroToJsonKeyStats(avroStats *AvroJsonKeyStats) *datapb.JsonKeyStats {
+	if avroStats == nil {
+		return nil
+	}
+	return &datapb.JsonKeyStats{
+		FieldID:                avroStats.FieldID,
+		Version:                avroStats.Version,
+		Files:                  avroStats.Files,
+		LogSize:                avroStats.LogSize,
+		MemorySize:             avroStats.MemorySize,
+		BuildID:                avroStats.BuildID,
+		JsonKeyStatsDataFormat: avroStats.JsonKeyStatsDataFormat,
+	}
+}
+
+// convertJsonKeyIndexMapToAvro converts protobuf map[int64]*JsonKeyStats to Avro array format.
+// Avro doesn't support maps with non-string keys, so we convert to array of entries.
+func convertJsonKeyIndexMapToAvro(indexMap map[int64]*datapb.JsonKeyStats) []AvroJsonKeyIndexEntry {
+	var entries []AvroJsonKeyIndexEntry
+	for fieldID, stats := range indexMap {
+		entries = append(entries, AvroJsonKeyIndexEntry{
+			FieldID: fieldID,
+			Stats:   convertJsonKeyStatsToAvro(stats),
+		})
+	}
+	return entries
+}
+
+// convertAvroToJsonKeyIndexMap converts Avro array back to protobuf map format.
+func convertAvroToJsonKeyIndexMap(entries []AvroJsonKeyIndexEntry) map[int64]*datapb.JsonKeyStats {
+	indexMap := make(map[int64]*datapb.JsonKeyStats)
+	for _, entry := range entries {
+		indexMap[entry.FieldID] = convertAvroToJsonKeyStats(entry.Stats)
+	}
+	return indexMap
+}
+
+// =============================================================================
+// Section 7: Avro Schema Definition
+// =============================================================================
+
+// getProperAvroSchema returns the Avro schema definition for ManifestEntry records.
+// This schema defines the complete structure of segment manifest files, including:
+//   - Basic segment metadata (ID, partition, level, channel, row count)
+//   - Binlog file references (insert, delete, stats, BM25 stats)
+//   - Index file references (vector indexes, text indexes, JSON key indexes)
+//   - Message queue positions (start and DML positions)
+//   - Storage version and sorting information
+//
+// Schema design notes:
+//   - Uses int64 for timestamps (Avro doesn't have uint64)
+//   - Uses arrays instead of maps (Avro maps require string keys)
+//   - Nested record types are defined inline and referenced by name
+//   - All fields are required (no null unions for simplicity)
+func getProperAvroSchema() string {
+	return `{
```

Comment:
what if we want to change the schema?
I think we should have some version in avro schema?